### ISSUE TITLE
Webcam not working

### ISSUE THREAD

**Issue by DreamProger** [bigbluebutton/bigbluebutton#6001]
Hi! I installed BBB HTML 5 client. All is well, But when i enabling webcam i giving a error.

Webcam is loading... and after sec 5-10 shows "webcam error". In console i see this: 
1022fe2a8ae52aa23b60a4e9c150efaf59f265c790.js?meteor_js_resource=true:147 Uncaught TypeError: Cannot read property 'peerConnection' of undefined
    at t.e [as stopWebRTCPeer] (22fe2a8ae52aa23b60a4e9c150efaf59f265c790.js?meteor_js_resource=true:147)
    at RTCPeerConnection.<anonymous> (22fe2a8ae52aa23b60a4e9c150efaf59f265c790.js?meteor_js_resource=true:147)

At same laptop  on official test server works well. I'm tried to understand what going bad. So i saw this:
1. When i'm clicking to enable webcam starts wss connection to domain/bbb-webrts-sfu.
2. Client sends websocket frames with WEBRTC offer
3. On official server server sends answer WEBRTC, on my server sends nothing. Server answers only on ping command.
4. My server disable webcam and its generate JS error. I wrote about it on top.

I tried:
1. Reinstall HTML5 BBB,
2. Reinstall Kurento,
3. Install TUNN server (i don't know how test TUNN server, so, i don't know well)

What i need to do to solve problem?

**Comment by DreamProger** [comment]
![image](https://user-images.githubusercontent.com/6420235/44539048-f53ee080-a724-11e8-9d6f-bd5b0d08b022.png)
No answer from BBB-WEBRTS-SFU

**Comment by DreamProger** [comment]
BigBlueButton Server 2.0.0-RC4 (1532)
                    Kernel version: 4.4.0-133-generic
                      Distribution: Ubuntu 16.04.5 LTS (64-bit)
                            Memory: 8134 MB

/var/www/bigbluebutton/client/conf/config.xml (bbb-client)
                Port test (tunnel): rtmp://edu.eiti.kz
                              red5: edu.eiti.kz
              useWebrtcIfAvailable: true

/opt/freeswitch/etc/freeswitch/sip_profiles/external.xml (FreeSWITCH)
                         websocket: :7443
                    WebRTC enabled: true

/etc/nginx/sites-available/bigbluebutton (nginx)
                       server name: edu.eiti.kz
                              port: 80, [::]:80
                              port: 443 ssl
                    bbb-client dir: /var/www/bigbluebutton

/var/lib/tomcat7/webapps/bigbluebutton/WEB-INF/classes/bigbluebutton.properties (bbb-web)
                      bbb-web host: edu.eiti.kz

/var/lib/tomcat7/webapps/demo/bbb_api_conf.jsp (API demos)
                               url: edu.eiti.kz

/var/www/bigbluebutton/check/conf/config.xml (client check)
                      client check: edu.eiti.kz

/usr/share/red5/webapps/bigbluebutton/WEB-INF/red5-web.xml (red5)
                  voice conference: FreeSWITCH

/usr/local/bigbluebutton/core/scripts/bigbluebutton.yml (record and playback)
                     playback host: edu.eiti.kz
                            ffmpeg: 4.0.2-1~16.04.york0.1

/usr/local/bigbluebutton/bbb-webrtc-sfu/config/default.yml (Kurento)
                        kurentoUrl: ws://edu.eiti.kz:8888/kurento
                         kurentoIp: 185.31.161.96
                    localIpAddress: 185.31.161.96
               recordScreenSharing: true
                     recordWebcams: true
                              Node: v8.11.4
                           mongoDB: v3.4.16


** Potential problems described below **
# IP does not match:
#                           IP from ifconfig: 185.31.161.96
#   /etc/nginx/sites-available/bigbluebutton: edu.eiti.kz
# Warning: API URL IPs do not match host:
#
#                                IP from ifconfig: 185.31.161.96
#  /var/lib/tomcat7/webapps/demo/bbb_api_conf.jsp: edu.eiti.kz


# Warning: The API demos are installed and accessible from:
#
#    https://edu.eiti.kz/demo/demo1.jsp
#
# These API demos allow anyone to access your server without authentication
# to create/manage meetings and recordings. They are for testing purposes only.
# If you are running a production system, remove them by running:
#
#    sudo apt-get purge bbb-demo

# Warning: The client self check is installed and accessible from:
#
#    https://edu.eiti.kz/check

**Comment by ffdixon** [comment]
Glad to hear its now working!

**Comment by ffdixon** [comment]
BigBlueButton comes with it's own version of Kurento, so you shouldn't need to install your own version. 
 Were you initially testing with the built-in version of Kurento with BigBlueButton, or had you installed Kurento 6.7.1 and then updated to 6.7.2?

**Comment by DreamProger** [comment]
@ffdixon, i installed BBB on new clean Ubuntu 16.04 server, and had problem - webcam is not working, after debugging BBB-WEBRTC NODE module i found error - Kurento server is not responding. Websocket connection don't starts, Kurento don't responding HTTP 101 to start WebSocket session. After i tried to reinstall Kurento from official kurento website. Their not updated tutors and i installed Kurento 6.7.1. Problem is not solved, kurento webserver don't response.
Previously i enabled UFW firewall to protect server.  I disabled UFW and rebooted server, then all works. After googling i found this - https://stackoverflow.com/questions/48557856/ufw-not-allowing-kurento-media-server-6-7-to-get-connected-through-ws-uri . I think it was some problem beetwen websocket server and ufw. I found 6.7.2 Kurento version on github and installed. Now BBB, Kurento and another works beatifull with enabled UFW firewall.
